en/test_and_ci/unit_tests.md

```diff
@@ -1,11 +1,34 @@
 # Unit Tests
 
-PX4 provides a simple base [Unittest-class](https://github.com/PX4/Firmware/blob/master/src/include/unit_test.h). Each developer is encouraged to write unit tests in the process of adding a new feature to the PX4 framework.
-
-## Writing a Test
-
-1. Create a new .cpp file within [tests](https://github.com/PX4/Firmware/tree/master/src/systemcmds/tests) with name **test_[description].cpp**. 
-1. Within **test_[description].cpp** include the base unittest-class `<unit_test.h>` and all files required to write a test for the new feature. 
+PX4 provides several methods for writing unit tests. Each developer is encouraged to write unit tests during all parts of development, including adding new features, fixing bugs, and refactoring.
+
+1. Unit tests with GTest - tests that have minimal, internal-only dependencies
+1. Functional tests with GTest - tests that depend on parameters and uORB messages
+1. SITL unit tests. This is for tests that need to run in full SITL. These tests are much slower to run and harder to debug, so it is recommended to use GTest instead when possible.
+
+## Writing a GTest Unit Test
+
+1. Unit tests should be arranged in three sections: setup, run, check results. Each test should test one very specific behavior or setup case, so if a test fails it is obvious what is wrong. Please try to follow these standards when possible.
+1. Copy and rename the [example unit test](https://github.com/PX4/Firmware/blob/master/src/modules/mc_att_control/AttitudeControl/AttitudeControlTest.cpp) to the directory the code to be tested is in.
```

Comment:
Will it be a separate example test? if it is this one then you would do something like 

Copy and rename the example unit test [AttitudeControlTest](https://github.com/PX4/Firmware/blob/master/src/modules/mc_att_control/AttitudeControl/AttitudeControlTest.cpp)  ...